### PR TITLE
Docs: Add transactional ID to the documentation

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Documentation

### Description

This PR updates documentation to match changes from PR #1048 

### Checklist

- [x] Update documentation

### CODE REVIEW

File: documentation/book/ref-kafka-user.adoc [hunk 1]
```diff
@@ -139,16 +139,17 @@ Use `\*` to allow or deny the operation from all hosts.
...
+Simple Authorization supports 4 different resource types:
 +
 * Topics
 * Consumer Groups
 * Clusters
+* Transactional IDs
 +
 The resource type can be specified in the `type` property.
-Use `topic` for Topics, `group` for Consumer Groups and `cluster` for clusters.
+Use `topic` for Topics, `group` for Consumer Groups, `cluster` for clusters, and `transactionId` for Transactional IDs.
```
Comment (ppatierno) [inline documentation/book/ref-kafka-user.adoc]: is that `transactionalId` and not `transactionId` ?

Comment (scholzj) [reply documentation/book/ref-kafka-user.adoc]: Dammit ... I checked it 5 times if I have them all correctly. Thanks for noticing.

**Comment by scholzj** [comment]
@ppatierno I think we need it. But it's beyond the scope of this PR. And it should be IMHO in the Kafka docs it self.

**Comment by ppatierno** [comment]
Agree

File: documentation/book/ref-kafka-user.adoc [hunk 2]
```diff
@@ -139,16 +139,17 @@ Use `\*` to allow or deny the operation from all hosts.
 The `host` field is optional and when not specified, the value `*` will be used as default.
 
 `resource`:: Specifies the resource for which does the rule apply.
```
Comment (laidan6000) [inline documentation/book/ref-kafka-user.adoc]: _Please change:_

Specifies the resource for which the rule applies.

File: documentation/book/ref-kafka-user.adoc [hunk 3]
```diff
@@ -139,16 +139,17 @@ Use `\*` to allow or deny the operation from all hosts.
...
-Simple Authorization supports 3 different resource types:
+Simple Authorization supports 4 different resource types:
```
Comment (laidan6000) [inline documentation/book/ref-kafka-user.adoc]: 4 > four

(As per IBM Style Guide: "In general, write in words numbers that are less than 10".)

File: documentation/book/ref-kafka-user.adoc [hunk 4]
```diff
@@ -139,16 +139,17 @@ Use `\*` to allow or deny the operation from all hosts.
...
 * Consumer Groups
 * Clusters
+* Transactional IDs
 +
 The resource type can be specified in the `type` property.
-Use `topic` for Topics, `group` for Consumer Groups and `cluster` for clusters.
+Use `topic` for Topics, `group` for Consumer Groups, `cluster` for clusters, and `transactionalId` for Transactional IDs.
 +
-Topic and Group resources additionally allow to specify the name of the resource for which the rule applies.
+Topic, Group, and Transactional ID resources additionally allow to specify the name of the resource for which the rule applies.
```
Comment (laidan6000) [inline documentation/book/ref-kafka-user.adoc]: _Reword as follows:_

Additionally, Topic, Group, and Transactional ID resources allow you to specify the name of the resource for which the rule applies.

**Pull request merged**